packages/headless/src/controllers/triggers/headless-query-trigger.ts

```diff
@@ -56,25 +56,39 @@ export function buildQueryTrigger(engine: SearchEngine): QueryTrigger {
 
   const getState = () => engine.state;
 
-  let previousQueryTrigger: string = getState().triggers.query;
-  let originalQuery: string = getState().query.q;
+  let originalQuery = '';
+  let newQuery = '';
 
   return {
     ...controller,
 
     subscribe(listener: () => void) {
       const strictListener = () => {
-        const hasChanged = previousQueryTrigger !== this.state.newQuery;
-        previousQueryTrigger = this.state.newQuery;
+        const currentTrigger = getState().triggers.query;
+        const currentQuery = getState().query.q;
 
-        if (hasChanged && this.state.newQuery) {
-          originalQuery = getState().query.q;
+        if (currentQuery === currentTrigger) {
+          return;
+        }
```

Comment:
Without this, there was an infinite loop. I could not test the infinite loop with UTs, so I will add Cypress tests in Atomic when the  query trigger is added in Atomic.